crates/stackable-operator/src/kvp/annotation/mod.rs

```diff
@@ -150,182 +43,91 @@ impl Annotation {
-    /// overwrite any existing annotation already present.
-    pub fn parse_insert(
-        &mut self,
-        annotation: impl TryInto<Annotation, Error = AnnotationError>,
-    ) -> Result<(), AnnotationError> {
-        self.0.insert(annotation.try_into()?.0);
-        Ok(())
-    }
-
-    /// Inserts a new [`Annotation`]. This function will overwrite any existing
-    /// annotation already present.
-    pub fn insert(&mut self, annotation: Annotation) -> &mut Self {
-        self.0.insert(annotation.0);
-        self
-    }
-
-    // This forwards / delegates associated functions to the inner field. In
-    // this case self.0 which is of type KeyValuePairs<T>. So calling
-    // Annotations::len() will be delegated to KeyValuePair<T>::len() without
-    // the need to write boilerplate code.
-    delegate! {
-        to self.0 {
-            /// Tries to insert a new [`Annotation`]. It ensures there are no duplicate
-            /// entries. Trying to insert duplicated data returns an error. If no such
-            /// check is required, use [`Annotations::insert`] instead.
-            pub fn try_insert(&mut self, #[newtype] annotation: Annotation) -> Result<(), AnnotationsError>;
-
-            /// Extends `self` with `other`.
-            pub fn extend(&mut self, #[newtype] other: Self);
-
-            /// Returns the number of labels.
-            pub fn len(&self) -> usize;
-
-            /// Returns if the set of labels is empty.
-            pub fn is_empty(&self) -> bool;
-
-            /// Returns if the set of annotations contains the provided
-            /// `annotation`. Failure to parse/validate the [`KeyValuePair`]
-            /// will return `false`.
-            pub fn contains(&self, annotation: impl TryInto<KeyValuePair<AnnotationValue>>) -> bool;
-
-            /// Returns if the set of annotations contains a label with the
-            /// provided `key`. Failure to parse/validate the [`Key`] will
-            /// return `false`.
-            pub fn contains_key(&self, key: impl TryInto<Key>) -> bool;
-
-            /// Returns an [`Iterator`] over [`Annotations`] yielding a reference to every [`Annotation`] contained within.
-            pub fn iter(&self) -> impl Iterator<Item = &KeyValuePair<AnnotationValue>>;
-
+pub type Annotations = KeyValuePairs<AnnotationValue>;
+
+/// Well-known annotations used by other tools or standard conventions.
+pub mod well_known {
+    /// Annotations applicable to Stackable Secret Operator volumes
+    pub mod secret_volume {
+        use crate::{
+            builder::pod::volume::SecretOperatorVolumeScope,
+            kvp::{Annotation, AnnotationError},
+        };
+
+        /// Constructs a `secrets.stackable.tech/class` annotation.
+        pub fn secret_class(secret_class: &str) -> Result<Annotation, AnnotationError> {
+            Annotation::try_from(("secrets.stackable.tech/class", secret_class))
```

Comment:
I'd argue the opposite, moving towards dropping the `consts` module entirely since you should be using the typed interface instead. It feels weird to have both a typed and stringly interface for the same thing.

Reply:
Especially when the label values are all hardcoded in the doccomments _anyway_.